frontend/app/models/search.rb

```diff
@@ -10,15 +10,23 @@ def self.for_type(repo_id, type, criteria)
   end
 
   # :context_criteria - added to criteria for building the query, but hidden from user
-  def self.all(repo_id, criteria, context_criteria = {})
+  # :sorting - ignore repo preferences and use provided sort for results (i.e. for typeaheads)
+  def self.all(repo_id, criteria, context_criteria = {}, sorting = nil)
     context_criteria.each do |k, v|
       raise "Not implemented" unless v.is_a? Array
       criteria[k] ||= []
       criteria[k] += v
     end
     build_filters(criteria)
     criteria["page"] = 1 if not criteria.has_key?("page")
-    criteria["sort"] ||= sort(criteria["type[]"] || (criteria["filter_term[]"] || []).collect { |term| ASUtils.json_parse(term)['primary_type'] }.compact)
+
+    if sorting
+      criteria["sort"] = sorting
+    else
+      criteria["sort"] ||= sort(criteria["type[]"] || (criteria["filter_term[]"] || []).collect { |term| ASUtils.json_parse(term)['primary_type'] }.compact)
+    end
+
+    Rails.logger.debug("sort criteria: #{criteria['sort']}")
```

Comment:
```suggestion

    Rails.logger.debug("sort criteria: #{criteria['sort']}")
```

Reply:
Since that original sort criteria (now in the `else` block) isn't put in place until we hit this point here, I think this is the reason it still may be necessary to pass that flag through, to avoid changing the search results for any other pre-existing code that depends on it.

Reply:
I think keeping the contents of the `else` block would have the same effect:
```
criteria["sort"] ||= sort(criteria["type[]"] || (criteria["filter_term[]"] || []).collect { |term| ASUtils.json_parse(term)['primary_type'] }.compact)
```
But I also think it might make more sense to add this to the `context_criteria` param, since that is intended for parts of the query that are determined by the context in the app UI rather than the selections of the user, and do not get returned to the user in views, etc. There is a test here that might be a good pattern for another ensuring that existing sort parameters don't get clobbered:
https://github.com/archivesspace/archivesspace/blob/master/frontend/spec/models/search_spec.rb

Reply:
That makes sense - I'll look into using `context_criteria`